Pin Dropper/ViewController.swift

```diff
@@ -10,30 +10,70 @@ import UIKit
 
 class ViewController: UIViewController, GMSMapViewDelegate {
 
+    var pins : [Pin] = []
+    
+    @IBOutlet weak var mapUIView: UIView!
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        var camera = GMSCameraPosition.cameraWithLatitude(-33.86,
-            longitude: 151.20, zoom: 6)
-        var mapView = GMSMapView.mapWithFrame(CGRectZero, camera: camera)
+        var camera = GMSCameraPosition.cameraWithLatitude(-33.883633,
+            longitude: 151.193927, zoom: 14)
+        var mapView = GMSMapView.mapWithFrame(CGRectMake(0, 0, self.mapUIView.frame.size.width, self.mapUIView.frame.size.height), camera: camera)
         mapView.myLocationEnabled = true
-        self.view = mapView
+        self.mapUIView.addSubview(mapView)
+        self.view.addSubview(mapUIView)
         mapView.delegate = self
-        
-        var marker = GMSMarker()
-        marker.position = CLLocationCoordinate2DMake(-33.86, 151.20)
-        marker.title = "Sydney"
-        marker.snippet = "Australia"
-        marker.map = mapView
     }
 
     override func didReceiveMemoryWarning() {
         super.didReceiveMemoryWarning()
         // Dispose of any resources that can be recreated.
     }
+    
+   
+    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject!) {
+        if (segue.identifier == "ToPinsList") {
+            var svc = segue.destinationViewController as! PinsListViewController;
+            
+            svc.passed = pins
+            
+        }
+    }
 
     
     func mapView(mapView: GMSMapView!, didTapAtCoordinate coordinate: CLLocationCoordinate2D) {
+        var address : String?
+        var long : CLLocationDegrees = coordinate.longitude
+        var lat : CLLocationDegrees = coordinate.latitude
+        var location = CLLocation(latitude: lat, longitude: long) //changed!!!
+
+        //var placemark :AnyObject
+        CLGeocoder().reverseGeocodeLocation(location, completionHandler: {(placemarks, error) -> Void in
+            if error != nil {
+                println("Reverse geocoder failed with error" + error.localizedDescription)
+                return
+            }
+            
+            if placemarks.count > 0 {
+                let pm = placemarks[0] as! CLPlacemark
```

Comment:
`pm` is an incorrect name. `placemark` is more appropriate

Also, `if let placemark = placemarks[0] as? CLPlacemark` is more safe. That way, if for any reason... the typecast fails, the app will not crash